core/src/test/groovy/noe/server/AS7UndeployTest.groovy

```diff
@@ -0,0 +1,97 @@
+package noe.server
+
+import noe.eap.server.as7.AS7Properties
+import org.junit.AfterClass
+import org.junit.BeforeClass
+import org.junit.Test
+
+import static org.junit.Assert.assertFalse
+import static org.junit.Assert.assertNotNull
+import static org.junit.Assert.assertTrue
+
+class AS7UndeployTest {
+
+    private static final String as7Dir = "test-as7"
+
+    private static File baseDir
+    private static File deploymentsDir
+
+    private static AS7 testServer
+
+    @BeforeClass
+    static void prepare() {
+        baseDir = File.createTempDir('noe', 'ServerAbstract')
+        assertNotNull(baseDir)
+        deploymentsDir = new File(baseDir, "${as7Dir}/standalone/deployments")
+        deploymentsDir.mkdirs()
+        assertNotNull(deploymentsDir)
+
+        //some random value, necessary for AS7 to instantiate (not optimal)
+        System.setProperty("eap.version", "7.0.0")
+        testServer = new AS7(baseDir.getAbsolutePath(), as7Dir)
+    }
+
+    @AfterClass
+    static void cleanUp() {
+        if (testServer != null) {
+            testServer.kill()
+        }
+        baseDir.deleteDir()
```

Comment:
@honza-kasik if basedir for whatever reason in prepare() is null and an exception is thrown, this will throw a npe, maybe causing some confusion in bug hunting. Consider using the https://groovy-lang.org/operators.html#_safe_navigation_operator to avoid the npe.

Reply:
Good point, check for null added.